pkg/buildpipeline/pipeline.go

```diff
@@ -862,7 +839,7 @@ func createPipelineTasks(stage *TransformedStage) []pipelinev1alpha1.PipelineTas
 		_, provider := findWorkspaceProvider(stage, stage.getEnclosing(0))
 		var previousStageNames []string
 		for _, previousStage := range findPreviousNonBlockStages(*stage) {
-			previousStageNames = append(previousStageNames, previousStage.Task.Name)
+			previousStageNames = append(previousStageNames, previousStage.PipelineTask.Name)
```

Comment:
We want the `PipelineTask.Name` to be more human-friendly - not the full CRD/DNS-eligible requirements for `Task.Name`. And yeah, the linking is by `PipelineTask`, not `Task`, which makes sense, since you can use the same `Task` multiple times in the same `Pipeline`.

Reply:
(not that we're using the same `Task` multiple times yet, but we could theoretically end up doing so in the future, and I think we want to have the `PipelineTask` name be different from the `Task.Name` it's referring to, for clarity purposes if nothing else)